### PR TITLE
auth: more and better thread names

### PR DESCRIPTION
**Pull request by Habbie**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/lua-record.cc [hunk 1]
```diff
@@ -168,6 +171,7 @@ class IsUpOracle
   }
   void checkThread()
   {
+    setThreadName("pdns/luaupcheck");
```
Comment (rgacogne) [inline pdns/lua-record.cc:174]: If I'm not mistaken, `std::async` doesn't guarantee that the task will be executed in a different thread (although it's very likely), and if it is the thread might be long-lived and reused for a different task (pool of threads). 
So perhaps we might want to execute `setThreadName("pdns/luaupcheck");` again after"joining" the futures, to be sure that the name is not misleading while we are sleeping?

Comment (Habbie) [reply pdns/lua-record.cc:174]: it's my understanding that the `::async` launch policy guarantees a thread (but that's more based on current implementations than specs), but your idea is very good, I'll do it.

Comment (rgacogne) [reply pdns/lua-record.cc:174]: > it's my understanding that the `::async` launch policy guarantees a thread 

`as if in a new thread of execution represented by a std::thread object. ` Yup, you are right!

Comment (Habbie) [reply pdns/lua-record.cc:174]: did it anyway - what if we change the policy later, etc.

**Review by rgacogne — APPROVED** [review]

**Pull request merged**